File: tutorias-inf-uva-api/src/main/java/es/uva/inf/tutorias/business/domain/models/HorarioTutorias.java
package es.uva.inf.tutorias.business.domain.models;

import java.io.Serializable;
import java.util.List;

public class HorarioTutorias implements Serializable {
	private static final long serialVersionUID = 1L;

	private Long id;
	private List<FranjaTutorias> franjasTutorias;
	private TipoHorario tipoHorario;
	private Profesor profesor;

	public HorarioTutorias() {
	}

	public Long getId() {
		return this.id;
	}

	public void setId(Long id) {
		this.id = id;
	}

	public List<FranjaTutorias> getFranjasTutorias() {
		return this.franjasTutorias;
	}

	public void setFranjasTutorias(List<FranjaTutorias> franjasTutorias) {
		this.franjasTutorias = franjasTutorias;
	}

	public TipoHorario getTipoHorario() {
		return this.tipoHorario;
	}

	public void setTipoHorario(TipoHorario tipoHorario) {
		this.tipoHorario = tipoHorario;
	}

	public Profesor getProfesor() {
		return profesor;
	}

	public void setProfesor(Profesor profesor) {
		this.profesor = profesor;
	}

}

File: tutorias-inf-uva-frontend/src/app/models/horario-tutorias.ts
import { FranjaTutorias } from "./franja-tutorias";
import { TipoHorario } from "./tipo-horario";

export class HorarioTutorias {
	id?: number;
	tipoHorario?: TipoHorario;
	franjasTutorias?: FranjaTutorias[];

	constructor(id?: number, tipoHorario?: TipoHorario, franjasTutorias?: FranjaTutorias[]) {
		if (id) {
			this.id = id;
		}

		if (tipoHorario) {
			this.tipoHorario = tipoHorario;
		}

		if (franjasTutorias) {
			this.franjasTutorias = franjasTutorias;
		}
	}
}